### PR TITLE
[ft-benchmark] some fixes for benchmark infra

### PR DESCRIPTION
**Pull request by MCJOHN974**
With this PR we will be finally able to run benchmark and send data to DB fully automatically

### CODE REVIEW

**Comment by MCJOHN974** [comment]
Whoops, we actually able to change some hardcoded values in `ft-benchmark-data-sender.py` to command line arguments. Will fix it before landing

File: scripts/ft-benchmark-data-sender.py [hunk 1]
```diff
@@ -89,24 +85,33 @@ def get_commit() -> tuple[str, datetime]:
...
     chdir(os.path.expanduser("~/nearcore/benchmarks/continous/db/tool"))
-    with tempfile.NamedTemporaryFile() as fp:
+    with tempfile.NamedTemporaryFile(mode="w", encoding='utf-8') as fp:
         json.dump(data, fp)
-        fp.close()
+        fp.flush()
+        os.fsync(fp.fileno()) 
         subprocess.run(f"cargo run -p cli -- insert-ft-transfer {fp.name}",
                        shell=True)
+        fp.close()
```
Comment (akashin) [inline scripts/ft-benchmark-data-sender.py:96]: I think you should be able to remove `flush` and `fsync` by moving this `close` call after `json.dump`.

File: scripts/ft-benchmark-data-sender.py [hunk 2]
```diff
@@ -89,24 +85,33 @@ def get_commit() -> tuple[str, datetime]:
...
+    DURATION = parser.parse_args().duration / 3600
+
     state_size = (int(
         subprocess.check_output(["du", "-s", "~/.near/localnet/node0/data"],
                                 stderr=subprocess.PIPE,
                                 shell=True).decode("utf-8").split()[0]) * 1024)
     processed_transactions = []
     time_begin = datetime.now()
     while True:
+        print(f"Data sender loop. Time elapsed: {(datetime.now() - time_begin).seconds} seconds")
```
Comment (akashin) [inline scripts/ft-benchmark-data-sender.py:114]: Let's use a `logging` module for this. It will automatically include the time in the log tine.

**Review by akashin — APPROVED** [review]

File: scripts/ft-benchmark-data-sender.py [hunk 3]
```diff
@@ -89,24 +85,33 @@ def get_commit() -> tuple[str, datetime]:
...
         subprocess.run(f"cargo run -p cli -- insert-ft-transfer {fp.name}",
                        shell=True)
+        fp.close()
 
 
 # TODO: send signal to this process if ft-benchmark.sh decided to switch neard to another commit.
 # add handling of this signal to this script
 if __name__ == "__main__":
+
+    parser = argparse.ArgumentParser(description="Process some integers.")
```
Comment (akashin) [inline scripts/ft-benchmark-data-sender.py:103]: Can we make description more descriptive? :)

File: scripts/run-ft-benchmark.py [hunk 4]
```diff
@@ -52,9 +52,19 @@ def main() -> None:
 
     args = parser.parse_args()
 
+    time_seconds = args.time
```
Comment (akashin) [inline scripts/run-ft-benchmark.py:55]: I wonder if we can use some library for this? E.g. there is `parse_timespan` in Locust: https://github.com/locustio/locust/blob/1fc3143f1f275c6b311c9cecfab151410a76467e/locust/util/timespan.py#L5

File: scripts/ft-benchmark-data-sender.py (hunk 2 above)
Comment (MCJOHN974) [reply scripts/ft-benchmark-data-sender.py:114]: Probably time from start of this loop is more informative than just time. But in general I added this just to see if this script started

File: scripts/ft-benchmark-data-sender.py (hunk 1 above)
Comment (MCJOHN974) [reply scripts/ft-benchmark-data-sender.py:96]: Unfortunately no.  Probably setting some options while creating tmp file object can help, but I spent a lot of time on this function and didn't find such options

Comment (MCJOHN974) [reply scripts/ft-benchmark-data-sender.py:96]: Error message was a bit unobvious, so I spent a time playing with tmp file options and place of `fp.close()`, before figured out, that it just don't flushing for some reason

Comment (akashin) [reply scripts/ft-benchmark-data-sender.py:96]: Can you provide steps to reproduce this? According to Python documentation, `close` is guaranteed to call `flush`: https://docs.python.org/3/library/io.html#io.IOBase.close

Comment (akashin) [reply scripts/ft-benchmark-data-sender.py:96]: You need to disable `delete_on_close` though: https://github.com/python/cpython/blob/cde976d85c6632a908dde1ff8695ac5cafd879b6/Lib/tempfile.py#L540

**Pull request merged**